modules/runner/src/main/java/org/apache/ignite/runner/internal/storage/DistributedConfigurationStorage.java

```diff
@@ -18,38 +18,86 @@
 package org.apache.ignite.runner.internal.storage;
 
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
 import org.apache.ignite.configuration.storage.ConfigurationStorage;
 import org.apache.ignite.configuration.storage.ConfigurationStorageListener;
 import org.apache.ignite.configuration.storage.Data;
 import org.apache.ignite.configuration.storage.StorageException;
+import org.apache.ignite.lang.util.SerializationUtils;
+import org.apache.ignite.metastorage.common.Cursor;
+import org.apache.ignite.metastorage.common.Entry;
+import org.apache.ignite.metastorage.common.Key;
 import org.apache.ignite.metastorage.internal.MetaStorageManager;
 
 public class DistributedConfigurationStorage implements ConfigurationStorage {
+    // TODO Distributed prefix. Will be replaced when ENUM with configuration type will be presented.
+    // https://issues.apache.org/jira/browse/IGNITE-14476
+    private static String DISTRIBUTED_PREFIX = "distributed";
+
     private final MetaStorageManager metaStorageMgr;
 
     public DistributedConfigurationStorage(MetaStorageManager metaStorageMgr) {
         this.metaStorageMgr = metaStorageMgr;
     }
 
-    @Override public Data readAll() throws StorageException {
-        return null;
+    /** Change listeners. */
+    private List<ConfigurationStorageListener> listeners = new ArrayList<>();
+
+    /** Storage version. */
+    private AtomicLong version = new AtomicLong(0);
```

Comment:
Why do we use atomic here?
The variable is modified from synchronized method, it means nothing issue with concurrency be here.
Use simple long variable.